<html><head>
<style> table, th, td {
  border: 1px solid black;
  border-collapse: collapse;
}th {
  background: lightblue;
}</style>
</head> <body><h1>SATD</h1><table><tr><th>satd id</th> <th>satd instance id</th>  <th>project</th> <th>committer name </th> <th> Commit Hash</th> <th>old comment</th> <th>New Comment</th> <th>resolution</th> <th>Method Signature</th> <th>Method Declaration</th> <th>Method Body</th> </tr><tr><td>2145</td> <td>-1422650922</td><td>apache/camel</td><td>Claus Ibsen</td><td>d247aa944e9ff39ee3127fc1bd0b02199b2f6c4e</td> <td>None</td> <td>but the assertion would be complete before hand and thus
the assertion was valid at the time given</td> <td>SATD_ADDED</td> <td>testNoAssertPeriodSecondMessageArrives()</td> <td>public void testNoAssertPeriodSecondMessageArrives() throws Exception</td> <td>
    MockEndpoint mock = getMockEndpoint("mock:result");
    mock.expectedMessageCount(1);
    template.sendBody("direct:a", "A");
    // this executor was bound to send a 2nd message
    Executor executor = Executors.newSingleThreadExecutor();
    executor.execute(new Runnable() {

        public void run() {
            try {
                Thread.sleep(2000);
            } catch (Exception e) {
            // ignore
            }
            template.sendBody("direct:a", "B");
        }
    });
    // but the assertion would be complete before hand and thus
    // the assertion was valid at the time given
    assertMockEndpointsSatisfied();
</td> </tr></table></body></html>